src/util/misc.rs

```diff
@@ -17,29 +17,59 @@
 //! Various utility functions
 
 use blockdata::opcodes;
-use util::iter::Pairable;
 use consensus::encode;
 
+/// Helper function to convert hex nibble characters to their respective value
+#[inline]
+fn hex_val(c: u8) -> Result<u8, encode::Error> {
+    let res = match c {
+        b'0' ... b'9' => c - '0' as u8,
+        b'a' ... b'f' => c - 'a' as u8 + 10,
+        b'A' ... b'F' => c - 'A' as u8 + 10,
+        _ => return Err(encode::Error::UnexpectedHexDigit(c as char)),
+    };
+    Ok(res)
+}
+
 /// Convert a hexadecimal-encoded string to its corresponding bytes
-pub fn hex_bytes(s: &str) -> Result<Vec<u8>, encode::Error> {
-    let mut v = vec![];
-    let mut iter = s.chars().pair();
-    // Do the parsing
-    iter.by_ref().fold(Ok(()), |e, (f, s)| 
-        if e.is_err() { e }
-        else {
-            match (f.to_digit(16), s.to_digit(16)) {
-                (None, _) => Err(encode::Error::UnexpectedHexDigit(f)),
-                (_, None) => Err(encode::Error::UnexpectedHexDigit(s)),
-                (Some(f), Some(s)) => { v.push((f * 0x10 + s) as u8); Ok(()) }
-            }
+pub fn hex_bytes(data: &str) -> Result<Vec<u8>, encode::Error> {
+    // This code is optimized to be as fast as possible without using unsafe or platform specific
+    // features. If you want to refactor it please make sure you don't introduce performance
+    // regressions (see benches/from_hex.rs).
```

Comment:
Oh, right, had to remove them since it didn't work with 1.14. I thought about adding a `benchtarget` feature, but then the bench deps have to be normal dependencies instead of dev-dependencies, which doesn't sound that good either.

Would you rather remove the comment or do the feature hack?

Reply:
I guess since we're doing it for bitcoin_hashes no reason not to do it here, too?

Reply:
Oh, are we? I didn't know, but makes sense.

Reply:
Yea, we have benchmarks for all the hashers in bitcoin_hashes under the "unstable" feature, https://github.com/rust-bitcoin/bitcoin_hashes/blob/master/Cargo.toml

Reply:
Fixed, but I called the feature `bench`.